aiohttp/streams.py

```diff
@@ -254,15 +254,23 @@ def end_http_chunk_receiving(self) -> None:
         if self._http_chunk_splits is None:
             raise RuntimeError("Called end_chunk_receiving without calling "
                                "begin_chunk_receiving first")
-        if not self._http_chunk_splits or \
-                self._http_chunk_splits[-1] != self.total_bytes:
-            self._http_chunk_splits.append(self.total_bytes)
```

Comment:
Previous code added zero here.

Reply:
Do is the whole difference in `or self.total_bytes == 0` clause?

Reply:
The problem in adding zero as the first item of the array. Yes, total_bytes can be zero when the function is called.